pkg/gce-pd-csi-driver/controller.go

```diff
@@ -531,15 +535,30 @@ func (gceCS *GCEControllerServer) ListVolumes(ctx context.Context, req *csi.List
 		klog.Warningf("ListVolumes requested max entries of %v, GCE only supports values <=500 so defaulting value back to 500", maxEntries)
 		maxEntries = 500
 	}
-	diskList, nextToken, err := gceCS.CloudProvider.ListDisks(ctx, maxEntries, req.StartingToken)
-	if err != nil {
-		if gce.IsGCEInvalidError(err) {
-			return nil, status.Error(codes.Aborted, fmt.Sprintf("ListVolumes error with invalid request: %v", err))
+	if maxEntries == 0 {
+		maxEntries = 500
```

Comment:
I don't think there's any reason to keep the 500 max entries, that was a GCE thing. I think we should we return all entries in a single page if maxEntries == 0.

Reply:
done. also removed the limit of 500 if maxEntries is >500